app/models/invoice_item.rb

```diff
@@ -45,6 +45,8 @@ class InvoiceItem < ActiveRecord::Base
 
   after_destroy :recalculate_invoice!
 
+  after_update :recalculate!, if: :saved_change_to_count_or_unit_cost?
```

Comment:
ich würde gerne vermeiden, in einem `after_`-hook einen Record zu speichern. Das kann leicht zur Callback-Hölle werden.

Könnte man nicht die internen Daten des InvoiceItem in einem `before_update` aktualisieren und dann nur das Aktualisieren der Rechnung im `after_udpate` anstossen, sofern nötig?